### PR TITLE
feat: Implemented reset method

### PR DESCRIPTION
**Pull request by ikrivosheev**
Hello! It will be very useful to have `reset` method in Lzxd. I need it in `rust-cab` crate.

### CODE REVIEW

**Comment by ikrivosheev** [comment]
@Lonami it will be very cool to release these changes

File: src/lib.rs [hunk 1]
```diff
@@ -371,6 +372,11 @@ impl Lzxd {
             Ok(view)
         }
     }
+
+    pub fn reset(&mut self) {
```
Comment (Lonami) [inline src/lib.rs]: If we want to release new public API, we should document it, even if the method seems straightforward.

File: src/lib.rs [hunk 2]
```diff
@@ -242,11 +243,11 @@ impl Lzxd {
     }
 
     /// Attempts to perform post-decompression E8 fixups on an output data buffer.
-    fn postprocess<'a>(
+    fn postprocess(
```
Comment (Lonami) [inline src/lib.rs:246]: I'm not sure why this changed. Was it an autofix from clippy perhaps?

Comment (ikrivosheev) [reply src/lib.rs:246]: I got from idea: `Explicit lifetimes given in parameter types where they could be elided `  and it looks logical)

File: src/lib.rs (hunk 1 above)
Comment (ikrivosheev) [reply src/lib.rs]: Done

File: src/lib.rs [hunk 3]
```diff
@@ -371,6 +372,13 @@ impl Lzxd {
             Ok(view)
         }
     }
+
+    /// Quickly resets the state.
+    /// This is equivalent to dropping this object and then creating a new one.
```
Comment (Lonami) [inline src/lib.rs]:
```suggestion
    /// Resets the decoder state.
    ///
    /// This is equivalent to calling [`Self::new`] with the same [`WindowSize`].
    /// [`WindowSize`]: enum.WindowSize.html
```

**Pull request merged**